### PR TITLE
Do away with the requirement of adding every flag in mount_gcsfuse

### PR DESCRIPTION
**Pull request by kislaykishore**
* Get the list of flags from the code-generated list and use that to automatically convert the persistent mount flags to appropriate format.

### Link to the issue in case of a bug fix.
https://b.corp.google.com/issues/368503898

### Testing details
1. Manual - NA
2. Unit tests - Tested with existing unit tests.
3. Integration tests - NA

### CODE REVIEW

File: tools/mount_gcsfuse/main.go [hunk 1]
```diff
@@ -72,89 +100,35 @@ func makeGcsfuseArgs(
...
+
 	// Deal with options.
 	for name, value := range opts {
-		switch strings.ReplaceAll(name, "-", "_") {
-		// Don't pass through options that are relevant to mount(8) but not to
-		// gcsfuse, and that fusermount chokes on with "Invalid argument" on Linux.
-		case "user", "nouser", "auto", "noauto", "_netdev", "no_netdev":
-
-		// Special case: support mount-like formatting for gcsfuse bool flags.
-		case "implicit_dirs",
```
Comment (ashmeenkaur) [inline tools/mount_gcsfuse/main.go:83]: Did you find any discrepancy between the flags supported in persistent mounting and the normal flags? Just want to make sure that there are no flags which should not be supported here.

Comment (kislaykishore) [reply tools/mount_gcsfuse/main.go:83]: I don't think there is any value in doing that. For instance, "foreground" is something that doesn't really seem to make any sense here (correct me if I'm wrong), but it's already there.

**Review by ashmeenkaur — APPROVED** [review]

File: tools/mount_gcsfuse/main.go (hunk 1 above)
Comment (kislaykishore) [reply tools/mount_gcsfuse/main.go:83]: I've added "o" flag as an exception to ensure that its behavior is the same as before.

File: tools/mount_gcsfuse/main.go [hunk 2]
```diff
@@ -72,89 +100,35 @@ func makeGcsfuseArgs(
 	device string,
 	mountPoint string,
 	opts map[string]string) (args []string, err error) {
+	var flagSet pflag.FlagSet
+	if err := cfg.BuildFlagSet(&flagSet); err != nil {
+		return nil, err
+	}
+
```
Comment (ashmeenkaur) [inline tools/mount_gcsfuse/main.go:107]: Nit: remove empty line

File: tools/mount_gcsfuse/main.go [hunk 3]
```diff
@@ -72,89 +100,40 @@ func makeGcsfuseArgs(
...
+	boolFlags, nonBoolFlags := flagTypes(&flagSet)
+	// Handle "o" by not treating it as a flag for persistent mounting.
+	nonBoolFlags = slices.DeleteFunc(nonBoolFlags, func(s string) bool {
+		return s == "o"
+	})
+
```
Comment (ashmeenkaur) [inline tools/mount_gcsfuse/main.go]: Nit: remove empty line

**Review by ashmeenkaur — APPROVED** [review]

**Pull request merged**